pkg/generators/markers.go

```diff
@@ -231,119 +231,189 @@ func ParseCommentTags(t *types.Type, comments []string, prefix string) (CommentT
 	return commentTags, nil
 }
 
-// lintArrayIndices attempts to prevent:
-// 1. Skipped indices
-// 2. Non-consecutive indices
-func lintArrayIndices(markerComments []string, prefix string) error {
+// ExtractCommentTags parses comments for lines of the form:
+//
+//	'marker' + "key=value".
+//
+// Values are optional; "" is the default.  A tag can be specified more than
+// one time and all values are returned.  If the resulting map has an entry for
+// a key, the value (a slice) is guaranteed to have at least 1 element.
+//
+// Similar to version from gengo, but this version support only allows one
+// value per key (preferring explicit array indices), supports raw strings
+// with concatenation, and limits the usable characters allowed in a key
+// (for simpler parsing).
+//
+// Assignments and empty values have the same syntax as from gengo. Raw strings
+// have the syntax:
+//
+//	'marker' + "key>value"
+//	'marker' + "key>value"
+//
+// Successive usages of the same raw string key results in concatenating each
+// line with `\n` in between. It is an error to use `=` to assing to a previously
+// assigned key
+// (in contrast to types.ExtractCommentTags which allows array-typed
+// values to be specified using `=`).
+var (
+	allowedKeyCharacterSet = `[:_a-zA-Z0-9\[\]\-]`
+	valueEmpty             = regexp.MustCompile(fmt.Sprintf(`^(%s*)$`, allowedKeyCharacterSet))
+	valueAssign            = regexp.MustCompile(fmt.Sprintf(`^(%s*)=(.*)$`, allowedKeyCharacterSet))
+	valueRawString         = regexp.MustCompile(fmt.Sprintf(`^(%s*)>(.*)$`, allowedKeyCharacterSet))
+)
+
+func extractCommentTags(marker string, lines []string) (map[string]string, error) {
+	out := map[string]string{}
+	lastKey := ""
+	lastIndex := -1
+	lastArrayKey := ""
+
 	var lintErrors []error
-	prevLine := ""
-	prevIndex := -1
-	for _, line := range markerComments {
+
+	for _, line := range lines {
 		line = strings.Trim(line, " ")
 
-		// If there was a non-prefixed marker inserted in between, then
-		// reset the previous line and index.
-		if !strings.HasPrefix(line, "+"+prefix) {
-			prevLine = ""
-			prevIndex = -1
-			continue
-		}
+		// Make sure last key gets reset if we `continue`
```

Comment:
```suggestion
		// Make sure last vars get reset if we `continue`
```

And move comment directly above `last*` vars?  (This comment did save me time when reading this)

Reply:
moved